src/NLog/Targets/FileTarget.cs

```diff
@@ -1727,63 +1776,124 @@ private bool ShouldDeleteOldArchives()
         /// <param name="fileName">File name to be written.</param>
         /// <param name="ev">Log event that the <see cref="FileTarget"/> instance is currently processing.</param>
         /// <param name="upcomingWriteSize">The size in bytes of the next chunk of data to be written in the file.</param>
-        /// <returns><see langword="true"/> when archiving should be executed; <see langword="false"/> otherwise.</returns>
-        private bool ShouldAutoArchive(string fileName, LogEventInfo ev, int upcomingWriteSize)
+        /// <returns>Filename to archive. If <c>null</c>, then nothing to archive.</returns>
+        private string ShouldAutoArchive(string fileName, LogEventInfo ev, int upcomingWriteSize)
```

Comment:
The names of the `Should...` methods are misleading. How about renaming them to `GetFileForArchiving...`?